pod/perlunicode.pod

```diff
@@ -1817,50 +1817,157 @@ through C<0x10FFFF>.)
+
+Vast sums of money have been lost to such attacks.  It became important
+to not fall for them, which involves detecting malformed text and taking
+appropriate action (or inaction).
+
+The Unicode REPLACEMENT CHARACTER (U+FFFD) is crucial to detecting and
+handling these.  It has no purpose other than to indicate it is a
+substitute for something else.  It is generally displayed as a white
+question mark on a dark background that is shaped like a diamond (a
+rectangle rotated 45 degrees).
+
+When a malformed string is encountered, the code processing it should
+substitute the REPLACEMENT CHARACTER for it.  There are now strict rules
+as to what parts get replaced.  You should never try to infer what the
+replaced part was meant to be.  To do so could be falling into an
+attacker's trap.
+
+Many of the attack vectors were not originally envisioned by Unicode's
+creators, nor by its implementers, such as Perl.  Rules about what were
+acceptable strings were originally laxer, tightened as the school of hard
+knocks dictated.
+
+Unfortunately, the Perl interpreter's methods of working with Unicode
+strings were developed when we too were naive about the possibility of
+attacks.  Because of concerns about breaking existing code and continued
+naivety about the consequences, there has been resistance to changing
+it, and so our implementation has lagged behind Unicode's requirements
+and recommendations.  But, over the years, various improvements have
+been added to minimize the issues.
+
+Therefore, it is important to use the latest version of the Perl
+interpreter when working with Unicode.  Not unil Perl v5.44 is it fully
+hardened against known attack vectors.  And who knows what new ideas
+clever atackers may come up with in the future, that we will have to
+change to counter.  (Although no new ones have become known recently
+that the Unicode Standard wasn't prepared for.)  And CPAN modules can
+easily lag behind the interpreter itself.
+
+If you are coding in pure Perl, you are pretty much at the mercy of how
+the underlying tools handle Unicode.  UTF-8 is how Unicode strings are
+stored internally by the interpreter.  And on most platforms, most
+inputs, such as files, will also be in UTF-8.  You should be reading
+those files via
+
+    binmode $fh, ":encoding(UTF-8)";
+
+Use of any other method can lead to attacks.  On some platforms, files
+encoded in UTF-16 prevail.  Use
+
+    binmode $fh, ":encoding(UTF-16)";
+
+on these.  See L<Encode> for more information.
+
+If you are coding in XS, you B<need> to be using functions in the
+C<utf8_to_uv> family when parsing a string encoded in UTF-8.  This
+avoids the pitfalls of earlier APIs.  See L<perlapi/utf8_to_uv>.
+One of those pitfalls is that instead of a REPLACEMENT CHARACTER, a NUL
+can be returned when a malformation is encountered.  This could
+conceivably lead to trojan strings where the second, trojan, part is
+hidden from code that is expecting a NUL-terminated string.
+
```

Comment:
Maybe this "improvement" is self-destructive and a bad idea to put in this wiki article, but someone could link to the 5-10 CVEs and RT tickets in the last 15 years about data smuggling with a null char in the middle of a byte array aka string.

Yes, 90% of current downloaded/indexed CPAN code is using encoding unaware getter setter Perl API methods. 20%-45% of tarballs, a UTF8/L1, along with custom written/malicious/insane/non-production/illogical PP script, will cause some kind of trivial waste of time to fix bug or severe business commercial corporate $ making losses bug. SEGVs, timeouts, 504 gateway timeout, CPU infinite loops. Remote code execution/priv escalation is probably paranoia. but taking down a bunch of rack servers until someone attaches perl5db.pl, yeah, thats $ loss.

Reply:
I'm trying to impress upon the reader that not upgrading to 5.44 when it comes out has a risk associated with it.   Now that I'm looking again at what got in to 5.42, that's probably good enough.  So I'm open to what the consensus here becomes

Reply:
I have no more opinion beyond what i said, this para can stay, it can be removed, it can grow longer with links. If nobody has a + or - opinion to add it can stay as is